app/src/gecko/java/org/mozilla/focus/web/LocalizedContentGecko.kt

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.web
+
+import android.content.Context
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.support.v4.util.ArrayMap
+import android.view.View
+
+import org.mozilla.focus.R
+import org.mozilla.focus.locale.Locales
+import org.mozilla.focus.utils.HtmlLoader
+import org.mozilla.focus.utils.SupportUtils
+import org.mozilla.gecko.GeckoSession
+
+import java.io.File
+import java.io.PrintWriter
+
+object LocalizedContentGecko {
+    // We can't use "about:" because webview silently swallows about: pages, hence we use
+    // a custom scheme.
+    val URL_ABOUT = "focus:about"
+    val URL_RIGHTS = "focus:rights"
+    val mplUrl = "https://www.mozilla.org/en-US/MPL/"
+    val trademarkPolicyUrl = "https://www.mozilla.org/foundation/trademarks/policy/"
+    val gplUrl = "gpl.html"
+    val trackingProtectionUrl = "https://wiki.mozilla.org/Security/Tracking_protection#Lists"
+    val licensesUrl = "licenses.html"
+
+    fun handleInternalContent(url: String, geckoSession: GeckoSession, context: Context): Boolean {
+        if (URL_ABOUT == url) {
+            loadAbout(geckoSession, context)
+            return true
+        } else if (URL_RIGHTS == url) {
+            loadRights(geckoSession, context)
+            return true
+        }
+        return false
+    }
+
+    /**
+     * Load the content for focus:about
+     */
+    private fun loadAbout(geckoSession: GeckoSession, context: Context) {
+        val resources = Locales.getLocalizedResources(context)
+
+        val substitutionMap = ArrayMap<String, String>()
+        val appName = context.resources.getString(R.string.app_name)
+        val learnMoreURL = SupportUtils.getManifestoURL()
+
+        var aboutVersion = ""
+        try {
+            val packageInfo = context.packageManager.getPackageInfo(context.packageName, 0)
+            aboutVersion = String.format("%s (Build #%s)", packageInfo.versionName, packageInfo.versionCode)
+        } catch (e: PackageManager.NameNotFoundException) {
+            // Nothing to do if we can't find the package name.
+        }
+
+        substitutionMap["%about-version%"] = aboutVersion
+
+        val aboutContent = resources.getString(R.string.about_content, appName, learnMoreURL)
+        substitutionMap["%about-content%"] = aboutContent
+
+        val wordmark = HtmlLoader.loadPngAsDataURI(context, R.drawable.wordmark)
+        substitutionMap["%wordmark%"] = wordmark
+
+        putLayoutDirectionIntoMap(substitutionMap, context)
+
+        val data = HtmlLoader.loadResourceFile(context, R.raw.about, substitutionMap)
+        val path = context.filesDir
+        val file = File(path, "about.html")
+        writeDataToFile(file, data)
+
+        geckoSession.loadUri(Uri.fromFile(file))
```

Comment:
I guess this is a workaround because GeckoView can't load HTML from a String yet? Is this something that we are going to get (cc @snorp)? If not then we should think about not re-generating this file if it already exists (however after an app update we would need to re-generate because it might be oudated).

Reply:
Exactly, it was a work around. Relevant Bugzilla issue for the methods we use for WebView to be added to GeckoView is https://bugzilla.mozilla.org/show_bug.cgi?id=1322579 but not sure if GV team is currently considering adding this.